File: test_anannas_api.py
import requests
import json

def test_anannas_api():
    print("🧪 Testing Anannas AI API: api.anannas.ai/v1")
    print("==========================================\n")
    
    base_url = "https://api.anannas.ai/v1"
    api_key = "sk-cr-27275ae349694731b095effe5abe64e3"
    
    headers = {
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json",
        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36"
    }
    
    # Test 1: Get models endpoint
    print("Test 1: GET /models")
    try:
        response = requests.get(f"{base_url}/models", headers=headers, timeout=15)
        print(f"  Status: {response.status_code}")
        print(f"  Content-Type: {response.headers.get('content-type')}")
        
        if response.status_code == 200:
            try:
                data = response.json()
                print(f"  ✅ SUCCESS: Models retrieved")
                print(f"  Total models: {len(data.get('data', []))}")
                print(f"  First 10 models: {', '.join([m.get('id', 'unknown') for m in data.get('data', [])[:10]])}")
                print(f"  Response: {json.dumps(data, indent=2)[:500]}...")
                return data
            except ValueError:
                print(f"  Response (text): {response.text[:300]}...")
                return response.text
        else:
            print(f"  ❌ FAILED: {response.text}")
            return None
            
    except Exception as e:
        print(f"  ❌ Error: {e}")
        return None
    
    print()
    
    # Test 2: Chat completions endpoint with the specific model from example
    print("Test 2: POST /chat/completions with openai/gpt-5-mini")
    chat_data = {
        "model": "openai/gpt-5-mini",
        "messages": [
            {"role": "user", "content": "Explain quantum computing"}
        ]
    }
    
    try:
        response = requests.post(
            f"{base_url}/chat/completions", 
            headers=headers, 
            json=chat_data, 
            timeout=30
        )
        
        print(f"  Status: {response.status_code}")
        print(f"  Content-Type: {response.headers.get('content-type')}")
        
        if response.status_code == 200:
            try:
                data = response.json()
                print(f"  ✅ SUCCESS: Chat completion received")
                print(f"  Response: {json.dumps(data, indent=2)[:800]}...")
                return data
            except ValueError:
                print(f"  Response (text): {response.text[:500]}...")
                return response.text
        else:
            print(f"  ❌ FAILED: {response.text}")
            return None
            
    except Exception as e:
        print(f"  ❌ Error: {e}")
        return None

def test_additional_models():
    """Test a few more models to see how many are working"""
    base_url = "https://api.anannas.ai/v1"
    api_key = "sk-cr-27275ae349694731b095effe5abe64e3"
    
    headers = {
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json"
    }
    
    # Test a few different models
    test_models = [
        "openai/gpt-4o",      # Popular model
        "openai/gpt-4",       # Standard GPT-4
        "claude-3-5-sonnet",  # Anthropic model
        "gemini-pro"          # Google model
    ]
    
    working_models = []
    failed_models = []
    
    for model in test_models:
        print(f"\nTesting {model}...")
        chat_data = {
            "model": model,
            "messages": [
                {"role": "user", "content": "Hello"}
            ],
            "max_tokens": 50  # Keep it short for testing
        }
        
        try:
            response = requests.post(
                f"{base_url}/chat/completions",
                headers=headers,
                json=chat_data,
                timeout=15
            )
            
            if response.status_code == 200:
                print(f"  ✅ WORKING")
                working_models.append(model)
            else:
                print(f"  ❌ FAILED: {response.status_code}")
                failed_models.append(model)
                
        except Exception as e:
            print(f"  ❌ ERROR: {e}")
            failed_models.append(model)
    
    print(f"\n📊 Summary:")
    print(f"Working models: {len(working_models)}")
    print(f"Failed models: {len(failed_models)}")
    
    if working_models:
        print(f"✅ Working: {', '.join(working_models)}")
    if failed_models:
        print(f"❌ Failed: {', '.join(failed_models)}")

if __name__ == "__main__":
    result = test_anannas_api()
    print('\n' + '='*50)
    if result:
        print('🎉 Anannas API Test: SUCCESS')
        # If we got models data, also test a few more models
        if isinstance(result, dict) and 'data' in result:
            print(f"\nTesting additional models from the list...")
            test_additional_models()
    else:
        print('❌ Anannas API Test: FAILED')

File: test_anannas_api.js
import fetch from 'node-fetch';

async function testAnannasAPI() {
    console.log('🧪 Testing Anannas AI API: api.anannas.ai/v1');
    console.log('==========================================\n');
    
    const baseUrl = 'https://api.anannas.ai/v1';
    const apiKey = 'sk-cr-27275ae349694731b095effe5abe64e3';
    
    const headers = {
        'Authorization': `Bearer ${apiKey}`,
        'Content-Type': 'application/json',
        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'
    };
    
    // Test 1: Get models endpoint
    console.log('Test 1: GET /models');
    try {
        const response = await fetch(`${baseUrl}/models`, { 
            method: 'GET', 
            headers: headers,
            timeout: 15000
        });
        
        console.log(`  Status: ${response.status} ${response.statusText}`);
        console.log(`  Content-Type: ${response.headers.get('content-type')}`);
        
        if (response.ok) {
            const data = await response.json();
            console.log('  ✅ SUCCESS: Models retrieved');
            console.log(`  Total models: ${data.data.length}`);
            console.log(`  First 10 models: ${data.data.slice(0, 10).map(m => m.id).join(', ')}`);
            console.log(`  Sample response: ${JSON.stringify(data, null, 2).substring(0, 500)}...`);
            return data;
        } else {
            console.log(`  ❌ FAILED: ${await response.text()}`);
            return null;
        }
    } catch (error) {
        console.log(`  ❌ Error: ${error.message}`);
        return null;
    }
    
    console.log();
}

async function testChatCompletions() {
    console.log('Test 2: POST /chat/completions');
    
    const baseUrl = 'https://api.anannas.ai/v1';
    const apiKey = 'sk-cr-27275ae349694731b095effe5abe64e3';
    
    const headers = {
        'Authorization': `Bearer ${apiKey}`,
        'Content-Type': 'application/json'
    };
    
    // Test with the exact example from the user
    const chatData = {
        "model": "openai/gpt-5-mini",
        "messages": [
            { "role": "user", "content": "Explain quantum computing" }
        ]
    };
    
    try {
        console.log('Testing with openai/gpt-5-mini...');
        const response = await fetch(`${baseUrl}/chat/completions`, {
            method: 'POST',
            headers: headers,
            body: JSON.stringify(chatData),
            timeout: 30000
        });
        
        console.log(`  Status: ${response.status} ${response.statusText}`);
        
        if (response.ok) {
            const data = await response.json();
            console.log('  ✅ SUCCESS: Chat completion received');
            console.log(`  Response: ${JSON.stringify(data, null, 2).substring(0, 800)}...`);
            
            if (data.choices && data.choices[0] && data.choices[0].message) {
                console.log('\n  🤖 AI Response:');
                console.log(`  ${data.choices[0].message.content.substring(0, 500)}...`);
            }
            
            return data;
        } else {
            console.log(`  ❌ FAILED: ${await response.text()}`);
            return null;
        }
    } catch (error) {
        console.log(`  ❌ Error: ${error.message}`);
        return null;
    }
}

// Run the tests
async function runTests() {
    const modelsResult = await testAnannasAPI();
    
    console.log('\n' + '='.repeat(50));
    
    if (modelsResult) {
        console.log('✅ Models endpoint working - 325 models available');
        
        // Now test chat completions
        const chatResult = await testChatCompletions();
        
        console.log('\n' + '='.repeat(50));
        if (chatResult) {
            console.log('🎉 Anannas API Test: COMPLETE SUCCESS');
            console.log('✅ Models endpoint: Working (325 models)');
            console.log('✅ Chat completions: Working');
        } else {
            console.log('❌ Anannas API Test: PARTIAL SUCCESS');
            console.log('✅ Models endpoint: Working (325 models)');
            console.log('❌ Chat completions: Failed/Timeout');
        }
    } else {
        console.log('❌ Anannas API Test: FAILED');
        console.log('❌ Models endpoint: Not working');
    }
}

runTests().catch(error => {
    console.error('Test execution failed:', error);
});